pkg/controllers/deployment/deployment_controller.go

```diff
@@ -135,6 +150,13 @@ func NewOAuthServerWorkloadController(
 }
 
 func (c *oauthServerDeploymentSyncer) PreconditionFulfilled(_ context.Context) (bool, error) {
+	if oidcAvailable, err := common.ExternalOIDCConfigAvailable(c.authLister, c.kasLister, c.kasConfigMapLister); err != nil {
+		return false, err
+	} else if oidcAvailable {
+		// the route is no longer a pre-requisite
+		return true, nil
```

Comment:
The route is gone. But we can still fulfill the precondition without the need for a deployment of the oauth-server?